src/NuGet.Clients/NuGet.Console/PowerConsoleToolWindow.cs

```diff
@@ -146,20 +145,22 @@ public override void OnToolWindowCreated()
         {
             base.OnToolWindowCreated();
 
-            // Register key bindings to use in the editor
-            var windowFrame = (IVsWindowFrame)Frame;
-            if (windowFrame != null)
-            {
-                // Set F1 help keyword
-                WindowFrameHelper.AddF1HelpKeyword(windowFrame, keywordValue: F1KeywordValuePmc);
-            }
-            var cmdUi = VSConstants.GUID_TextEditorFactory;
-            windowFrame.SetGuidProperty((int)__VSFPROPID.VSFPROPID_InheritKeyBindings, ref cmdUi);
-
             // start a task when VS us idle and dont await it immediately
-            _loadTask = ThreadHelper.JoinableTaskFactory.StartOnIdle(
+            _loadTask = NuGetUIThreadHelper.JoinableTaskFactory.StartOnIdle(
                 async () =>
                 {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+                    // Register key bindings to use in the editor
+                    var windowFrame = (IVsWindowFrame)Frame;
```

Comment:
wouldn't this just throw an InvalidCastException if it couldn't cast? use ```Frame as IVsWindowFrame``` instead.

Reply:
no, since this is always a Window Frame. Besides this is the existing code